### ISSUE TITLE
Cannot modify visibility of private method?

### ISSUE THREAD

**Issue by facboy** [raphw/byte-buddy#1728]
Is the following not supposed to work (in an `AgentBuilder`)?

```java
return dynamicTypeBuilder
        .method(methodMatcher)
        .intercept(SuperMethodCall.INSTANCE)
        .transform(Transformer.ForMethod.withModifiers(Visibility.PROTECTED));
```

It's a bit hard to follow but when debugging it is applied to the transformed `MethodDescriptor` but this is then added to a `Prepared.Entry` which has its own `visibility` field. The `visibility` field gets its value from the original `methodDescriptor` and ignores modifiers on the transformed `MethodDescriptor`.

**Comment by raphw** [comment]
Have a look at `ModifierAdjustment` which can be applied as a decorator (without defining an interception). I think that is what you actually want to do.

**Comment by facboy** [comment]
hmm...so at runtime, once adjusted, i can see that the modifier has changed, but it still doesn't do quite what i wanted, so i guess i'll ask another question:

is it possible to change a method from `private` to `protected`, and then have (existing) subclasses override that method?

**Comment by raphw** [comment]
That should be possible, yes. This does however assume that both methods have the same signature.

What are you observing?

**Comment by facboy** [comment]
I can see with reflection that the modifier is changed, but the calling methods just invoke the "previously `private` but now `protected`" method directly and ignore the overriding methods in the sub-classes.

**Comment by raphw** [comment]
I assume that the method is already linked. What happens when you
retransform those classes without any changes?

Christopher Ng ***@***.***> schrieb am So., 24. Nov. 2024,
11:58:

**Comment by raphw** [comment]
For example: if a class path scanner looks for classes, this might trigger their loading.
